### DISCUSSION TITLE
Newb: How to create/update rows from REST API Get?

### DISCUSSION THREAD

**Discussion by michaellis** [Budibase/budibase#5898] [Help]
Hello.  I'm very new to all of this.  My apologies in advanced.
I want to use a cron task in an automation to create/update rows using an external rest api get call.
In this example I am working with some customer data.

I created two datasources a BudibaseDB and REST API.

When I test the REST API I get results like this
![image](https://user-images.githubusercontent.com/4265847/168476920-a4786c08-9e2f-4cce-8642-7ba995cb6934.png)

The JSON example looks like (trimmed some additional rows.
{
  "id": "5bb196da-fdcf-4950-b848-db8c81ab29ba",
  "code": "ASSROG",
  "name": "Assembled Products Corp.",
  "status": "Active",
  "type": "Automotive",
  "note": "",
  "supplierCode": "",
  "otherCustomerCode": "",
 ...
}

I created an automation with a CRON task, and then the call to the external data source.
![image](https://user-images.githubusercontent.com/4265847/168477025-fee4ff27-aa63-4312-bfa0-8f71184bd8cb.png)
![image](https://user-images.githubusercontent.com/4265847/168477112-1fe8c68e-302b-4e33-af97-7a7ddff4fe5a.png)

I wanted to loop over the returned results and call create or update rows.
![image](https://user-images.githubusercontent.com/4265847/168477465-1fa4ff31-f6cd-44f3-b80b-7d54e7957654.png)

It's not working... I think because the loop is returning strings instead of an array?
![image](https://user-images.githubusercontent.com/4265847/168477516-7676db80-f78d-4e88-82a1-b0675d9ad73b.png)

Do I need to transform the data before I call Create Row or Update Row?
Is the syntax of the Create Row field mapping correct?  i.e.  {{ loop.currentItem.id }}

Thanks again for the help.

Best regards,
Michael

**Comment by melohagan** [comment]
Hi @michaellis 

Thank you very much for your detailed question. You have highlighted a bug that I have raised here: https://github.com/Budibase/budibase/issues/5912

You have done everything right, but as you can see in the **JSON Visualizer**, an object is being returned from the External Data Connector, which is a bug because it should be an array. Your REST query is correct.
